lighthouse-core/gather/driver.js

```diff
@@ -71,15 +71,16 @@ class Driver {
 
   online = true;
 
-  // eslint-disable-next-line no-invalid-this
-  fetcher = new Fetcher(this);
-
   // eslint-disable-next-line no-invalid-this
   executionContext = new ExecutionContext(this);
 
   // eslint-disable-next-line no-invalid-this
   defaultSession = this;
 
+  /** @type {Fetcher} */
+  // eslint-disable-next-line no-invalid-this
+  fetcher = new Fetcher(this);
```

Comment:
Need `executionContext` and `defaultSession` to be defined for the Fetcher constructor.

Reply:
add a comment to that effect in the code?

Reply:
Now that `defaultSession` and `executionContext` are passed in individually, I feel it's pretty obvious on its own.

Reply:
Yes 100% agreed. It's not necessary with the new approach 🎉  :)